### ISSUE TITLE
Installing Wild Web Developer tools forces update of JustJ and Platform to latest version

### ISSUE THREAD

**Issue by sratz** [eclipse-wildwebdeveloper/wildwebdeveloper#1380]
Follow-up on #938.

Wild Web Developer uses the same updatesite for both the RCP product and the regular update site.

This is a problem, as the RCP product update site also contains transitive dependencies such as JustJ which are undesired in the 'other RCP product + install wild web as plugin' use-case.

An attempt was made in #1238 to filter out these dependencies, but this actually made things worse:

While it does result in the dependencies to be gone from the updatesite, the updatesite now contains **references** to the dependencies:
https://github.com/vrubezhny/wildwebdeveloper/blob/62d007dcfaf17c69cdf84f15ed8ae03952482245/repository/category.xml#L84-L85

This results

1) in these referenced repositories to be added to users' 'Available Update Sites' by p2, even when the check-for-update operation is canceled:
- Before
![image](https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/assets/14908423/813bf58a-8936-4982-a0f1-2a8567e06fab)

- Check for updates
![image](https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/assets/14908423/43301271-7f1a-49e7-8936-7cbcba124ed2)

- Press **Cancel** (!) and check update sites again:
![image](https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/assets/14908423/2f7d2d9f-8f33-4b16-8b50-67755aaa2f65)

2) What's worse, these two referenced repositories are not static updatesites, but dynamic ones (`/latest`). This actually causes an update of the underlying Platform and JustJ to potentially unwanted, newer versions.

I think using `<repository-reference>` in category.xml is a bad idea and should be removed.

Instead, the update-site should be split in two: one for the actual RCP product including the platform and JustJ dependencies, and one for the pure wild web developer features.

**Comment by akurtakov** [comment]
So to clarify what happens here: 
* Download/install EPP package - it has Java 17
* Check for updates and Java 21 gets installed
Do I get it correct?

**Comment by sratz** [comment]
Yes,

- https://download.eclipse.org/technology/m2e/releases/latest
  contains repository references
  - https://download.eclipse.org/eclipse/updates/4.28/
  - https://download.eclipse.org/lsp4e/releases/0.23.0/
  - https://download.eclipse.org/lsp4j/updates/releases/0.21.0/
  - https://download.eclipse.org/tm4e/releases/0.8.1/
  - https://download.eclipse.org/wildwebdeveloper/releases/1.3.0/
    contains repository references
    - https://download.eclipse.org/eclipse/updates/latest
    - https://download.eclipse.org/justj/jres/latest

See detailed steps here:
https://github.com/eclipse-equinox/p2/discussions/367#discussioncomment-7432337

**Comment by sratz** [comment]
I'd argue that -- let's call them 'platform dependencies' -- such as platform itself and justJ should be included in the update site of WWD like this.

If anything, an update site should point to *fixed* versions of such dependencies.

**Comment by OLibutzki** [comment]
I just want to concretize to critical this issue is.

Our current situation:
We have been using https://download.eclipse.org/technology/m2e/releases/latest to install m2e in an  Eclipse 4.30 (2023-12)  release.

After Eclipse 4.21 (2024-03) has been released and an update in the Eclipse IDe has been performed, we were not able to open Java files anymore. Here is the stacktrace:
```
java.lang.NoClassDefFoundError: org/eclipse/jface/text/rules/FastPartitioner
	at org.eclipse.jdt.internal.ui.text.JavaPartitionerManager.createDocumentPartitioner(JavaPartitionerManager.java:52)
	at org.eclipse.jdt.ui.text.JavaTextTools.createDocumentPartitioner(JavaTextTools.java:307)
	at org.eclipse.jdt.ui.text.JavaTextTools.setupJavaDocumentPartitioner(JavaTextTools.java:383)
	at org.eclipse.jdt.internal.ui.javaeditor.JavaDocumentSetupParticipant.setup(JavaDocumentSetupParticipant.java:39)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBufferManager$1.run(ResourceTextFileBufferManager.java:151)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBufferManager.createEmptyDocument(ResourceTextFileBufferManager.java:165)
	at org.eclipse.core.internal.filebuffers.ResourceTextFileBuffer.initializeFileBufferContent(ResourceTextFileBuffer.java:235)
	at org.eclipse.core.internal.filebuffers.ResourceFileBuffer.create(ResourceFileBuffer.java:247)
	at org.eclipse.core.internal.filebuffers.TextFileBufferManager.connect(TextFileBufferManager.java:116)
	at org.sonarlint.eclipse.core.internal.resources.DefaultSonarLintFileAdapter.getDocument(DefaultSonarLintFileAdapter.java:66)
	at org.sonarlint.eclipse.core.internal.jobs.AsyncServerMarkerUpdaterJob.updateMarkers(AsyncServerMarkerUpdaterJob.java:76)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2453)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2473)
	at org.sonarlint.eclipse.core.internal.jobs.AsyncServerMarkerUpdaterJob.doRun(AsyncServerMarkerUpdaterJob.java:52)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractSonarJob.run(AbstractSonarJob.java:37)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.ClassNotFoundException: org.eclipse.jface.text.rules.FastPartitioner cannot be found by org.eclipse.jdt.core.manipulation_1.21.0.v20240221-0406
	... 17 more
```
As one can see: Because of wildwebdeveloper adding https://download.eclipse.org/eclipse/updates/latest as p2 repository, jdt got updated to the 4.31 version. Unfortunately this update is incomaptible with the 4.30 JFace version.

It took us hours to investigate this issue until we ended up in the content.xml of wildwebdeveloper P2 repository.

**Comment by mickaelistria** [comment]
Anyone who is willing to provide a fix to improve that is more than welcome.
At the moment, none of the maintainer has plan to fix it; but we'd make highest priority to review incoming fixes as pull requests.

**Comment by OLibutzki** [comment]
I am not sure which direction to solve this issue is the best one.

Is it an option to omit the repository-references (https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/blob/master/repository/category.xml#L87) entirely?

**Comment by mickaelistria** [comment]
> Is it an option to omit the repository-references (https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/blob/master/repository/category.xml#L87) entirely?

That's worth trying.

**Comment by OLibutzki** [comment]
> That's worth trying.

Here we go: https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/pull/1491

**Comment by mickaelistria** [comment]
Let's close then with #1491 being merged. Note that at this stage, the best next step is probably to remove the product from the build/repo.

**Comment by merks** [comment]
This continues to cause problems.  In this case, users can update EPP 2024-03 to EPP 2024-06 without actually updating their JRE and then the updated product simply fails to start:

https://github.com/eclipse-equinox/p2/issues/508

**Comment by mickaelistria** [comment]
This should be fixed with a new release. I'm preparing one now.

**Comment by merks** [comment]
That sounds great.

**Comment by mickaelistria** [comment]
Released Wild Web Developer 1.3.4 and submitted https://github.com/eclipse-simrel/simrel.build/pull/335
